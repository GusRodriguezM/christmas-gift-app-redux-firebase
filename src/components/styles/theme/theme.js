const theme = {
    colors: {
        black: '#000000',
        loginButton: '#4285f4',
        disabledButton: '#999595',
        red: '#C30F16',
        white: '#FFFFFF',
        list: '#c8e4e4',
        navbar: '#e8ceac',
        boxShadow: 'rgba(9, 9, 9, 0.23)',
    }
}

export default theme;