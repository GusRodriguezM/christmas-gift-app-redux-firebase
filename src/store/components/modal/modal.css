.modal {
    position: fixed;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.5)    ;
    display: flex;
    align-items: center;
    justify-content: center;
}

.modal__content {
    width:  40%;
    height: 90%;
    background-color: #c8e4e4;
}

.modal__content--header, .modal__content--footer {
    padding: 10px;
    height: 10%;
}

.modal__content--title {
    margin: 0;
}

.modal__content--body {
    padding: 10px;
    border-top: 1px solid #c8e4e4;
    border-bottom: 1px solid #c8e4e4;
    height: 80%;
}