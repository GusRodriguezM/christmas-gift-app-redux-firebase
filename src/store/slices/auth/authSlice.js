import { createSlice } from '@reduxjs/toolkit';

export const authSlice = createSlice({
  name: 'auth',
  initialState: {
    status: 'checking', //options: checking, non-authenticated, authenticated
    uid: null,
    email: null,
    userName: null,
    photoURL: null,
    errorMessage: null,
    logged: false
  },
  reducers: {
    login: (state, { payload }) => {
        state.status = 'authenticated';
        state.uid = payload.uid;
        state.email = payload.email;
        state.userName = payload.displayName;
        state.photoURL = payload.photoURL;
        state.errorMessage = payload.errorMessage;
    },
    logout: (state, { payload }) => {
        state.status = 'non-authenticated';
        state.uid = null;
        state.email = null;
        state.userName = null;
        state.photoURL = null;
        state.errorMessage = payload.errorMessage;
    },
    checkingUserCredentials: (state, action) => {
      state.status = 'checking';
    },
  },
});

export const { login, logout, checkingUserCredentials } = authSlice.actions;