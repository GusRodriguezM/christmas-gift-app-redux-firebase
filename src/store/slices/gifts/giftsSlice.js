import { createSlice } from '@reduxjs/toolkit';

export const giftsSlice = createSlice({
  name: 'gifts',
  initialState: {
    gifts: [
      {
        name: 'Stuffed Bear',
        quantity: 1,
        image: 'https://images.unsplash.com/photo-1556012018-50c5c0da73bf?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=870&q=80',
        person: 'Janet',
        price: 10,
      },
      {
        name: 'Xbox Series S',
        quantity: 1,
        image: 'https://images.unsplash.com/photo-1605901309584-818e25960a8f?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1019&q=80',
        person: 'Jorge',
        price: 20,
      },
      {
        name: 'Guitar',
        quantity: 1,
        image: 'https://images.unsplash.com/photo-1516924962500-2b4b3b99ea02?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=870&q=80',
        person: 'Victor',
        price: 30,
      }
    ]
  },
  reducers: {
    addGift: (state) => { },
  },
});

export const { addGift } = giftsSlice.actions;