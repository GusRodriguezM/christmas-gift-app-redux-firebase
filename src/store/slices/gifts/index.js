export * from './giftsSlice';
export * from './thunks';