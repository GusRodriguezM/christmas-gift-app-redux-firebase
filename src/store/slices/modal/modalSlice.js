import { createSlice } from '@reduxjs/toolkit';

export const modalSlice = createSlice({
  name: 'modal',
  initialState: {
    open: false,
    option: ''
  },
  reducers: {
    openModal: (state) => {
        state.open = true;
    },
    closeModal: (state) => {
        state.open = false;
    },
    setOption: (state, action) => {
      state.option = action.payload;
    }
  },
});

export const { openModal, closeModal, setOption } = modalSlice.actions;