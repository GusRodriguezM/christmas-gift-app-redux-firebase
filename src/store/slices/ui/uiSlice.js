import { createSlice } from '@reduxjs/toolkit';

export const uiSlice = createSlice({
    name: 'modal',
    initialState: {
        msgError: ''
    },
    reducers: {
        setErrorMessage: (state, {payload}) => {
            state.msgError = payload;
        },
        removeErrorMessage: (state) => {
            state.msgError = null;
        },
    },
});

export const { setErrorMessage, removeErrorMessage } = uiSlice.actions;