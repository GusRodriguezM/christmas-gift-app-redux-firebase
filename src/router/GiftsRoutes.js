import React from 'react';
import { Navigate, Route, Routes } from 'react-router-dom';
import { GiftScreen } from '../components/gifts/GiftScreen';
import { Navbar } from '../components/navbar/Navbar';

export const GiftsRoutes = () => {
    return (
        <>
            <Navbar />

            <Routes>
                <Route path="gifts" element={<GiftScreen />} />

                <Route path="/*" element={<Navigate to="/gifts"/> } />
            </Routes>
        </>
    )
}
