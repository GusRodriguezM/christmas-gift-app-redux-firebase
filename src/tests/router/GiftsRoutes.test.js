import { configureStore } from "@reduxjs/toolkit";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import theme from "../../components/styles/theme/theme";
import { GiftsRoutes } from "../../router/GiftsRoutes";
import { authSlice } from "../../store/slices/auth/authSlice";
import { giftsSlice } from "../../store/slices/gifts/giftsSlice";
import { modalSlice } from "../../store/slices/modal/modalSlice";
import { uiSlice } from "../../store/slices/ui/uiSlice";
import { authenticatedState } from "../fixtures/authFixtures";

describe('Tests in the <GiftsRoutes />', () => {

    test('Should render the component correctly and match with the snapshot', () => {

        //Mocking the store with some states and a default auth state
        const store = configureStore({
            reducer: {
                auth: authSlice.reducer,
                ui: uiSlice.reducer,
                gifts: giftsSlice.reducer,
                modal: modalSlice.reducer
            },
            preloadedState: {
                auth: authenticatedState
            }
        });

        //Rendering the component
        const { container } = render(
            <Provider store={ store } >
                <MemoryRouter initialEntries={['/gifts']}>
                    <ThemeProvider theme={ theme } >
                        <GiftsRoutes />
                    </ThemeProvider>
                </MemoryRouter>
            </Provider>
        );

        //Expecting that the rendered component matches with the snapshot
        expect( container ).toMatchSnapshot();

        //Then we just expect the elements of the component exists
        //Expecting the link in the navbar exists
        expect( screen.getByRole('link', { name: 'Christmas Gift App' }) ).toBeTruthy();

        //Expecting the number of the buttons to be 5 
        /**
         * 1. Logout
         * 2. Add Gift
         * 3. Visualize
         * 4. Print
         * 5. Delete All
         */
        expect( screen.getAllByRole('button').length ).toBe(5);
        
        //Expecting to have to h tags in the component
        expect( screen.getAllByRole('heading').length ).toBe(2);
        
        //Expecting that the name of the user appears in the navbar
        expect( screen.getByText(`: ${store.getState().auth.userName}`) ).toBeTruthy();

    });

});