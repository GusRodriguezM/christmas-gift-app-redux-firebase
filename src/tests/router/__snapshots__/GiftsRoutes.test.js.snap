// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in the <GiftsRoutes /> Should render the component correctly and match with the snapshot 1`] = `
<div>
  <nav
    class="sc-dmRaPn igMnyy"
  >
    <a
      class="sc-kgflAQ jdpQcI"
      href="/"
    >
      Christmas Gift App
    </a>
    <div
      class="sc-fLlhyt bTbnoI"
    >
      <span
        class="sc-hHLeRK ikEMyq"
      >
        <i
          class="fa-solid fa-user-astronaut"
        />
        : Jon Doe
      </span>
      <button
        class="sc-bczRLJ glPbkZ"
      >
        <span>
          Logout
        </span>
        <i
          class="fa-solid fa-right-from-bracket"
        />
      </button>
    </div>
  </nav>
  <main
    class="sc-dIouRR fRVzPB"
  >
    <h1>
      Gifts:
    </h1>
    <button
      class="sc-bczRLJ sc-gsnTZi glPbkZ hbpwPw"
    >
      <span>
        Add Gift
      </span>
      <i
        class="fa-solid fa-circle-plus"
      />
    </button>
    <div
      class="sc-crXcEl cMhrjf"
    >
      <div
        class="sc-fEOsli jmjICg"
      >
        <div
          class="sc-idiyUo gSzZLM"
        >
          <h1>
            Total: $
            0
          </h1>
        </div>
      </div>
    </div>
    <div
      class="sc-kDDrLX sc-iqcoie jBGpND cLiSAZ"
    >
      There is no gifts in your list. Please start adding some!
    </div>
    <span
      class="sc-hHLeRK ikEMyq"
    >
      <h1>
        Total: 
        0
      </h1>
    </span>
    <div
      class="sc-gKXOVf VFXNc"
    >
      <button
        class="sc-bczRLJ sc-gsnTZi glPbkZ hbpwPw"
      >
        <span>
          Visualize
        </span>
        <i
          class="fa-solid fa-eye"
        />
      </button>
      <button
        class="sc-bczRLJ sc-gsnTZi glPbkZ hbpwPw"
      >
        <span>
          Print
        </span>
        <i
          class="fa-solid fa-print"
        />
      </button>
      <button
        class="sc-bczRLJ sc-gsnTZi glPbkZ hbpwPw"
      >
        <span>
          Delete All
        </span>
        <i
          class="fa-solid fa-dumpster"
        />
      </button>
    </div>
  </main>
</div>
`;
